### ISSUE TITLE
VSO Calendar - Events - Categories Colors 

### ISSUE THREAD

**Issue by lizrs** [Microsoft/vsts-team-calendar#154]
How can you change the colors of the Events /Categories in the calendar? For example: I'm trying to show all the work items of each team in a different color and one color for the org items.  All Teams event = Green color, Team A = Purple, Team B = Red, etc

**Comment by Sarmiite** [comment]
Hi! When we could expect to have this done? It's realllly required, because items are getting random colors which are really confusing users.

**Comment by willsmythe** [comment]
We are in the process of prioritizing feature for the next update. This is on the list.

**Comment by AlexKnowles** [comment]
Any updates on the priority of this feature / when it will be added?

**Comment by Tamra** [comment]
Any idea when this feature will be available?

**Comment by pabrams** [comment]
I've implemented something like what you want in my private fork.  I'll submit a PR, if my organization ever signs the paperwork to let me do that...

**Comment by zkarj735** [comment]
Still no movement on this? I only need four categories and it has chosen only two colours, each shared between two categories. As I cannot remove categories nor set the colours and the chosen two are a pale blue and pale green, everything looks the same! They may as well all be monochrome.

This shouldn't be an enhancement because it is a usability nightmare, particularly if the system cannot even choose different colours.

**Comment by zkarj735** [comment]
I have since discovered that the root cause of this problem is the whole category system in the calendar is seriously broken. It needs tearing down and rebuilding from scratch. I honestly don't know how it ever shipped.

**Comment by pabrams** [comment]
> I have since discovered that the root cause of this problem is the whole category system in the calendar is seriously broken. It needs tearing down and rebuilding from scratch. I honestly don't know how it ever shipped.

I dont think its quite _that_ bad - it does work, for yhe most part.  Its just no frills.  You get random colors. And thy dont persist month to month.  But they are at least unique, per catwgory

**Comment by pabrams** [comment]
> Could you please provide an update on this request?

Update: its not implemented, yet.

**Comment by zkarj735** [comment]
> I dont think its quite _that_ bad - it does work, for yhe most part. Its just no frills. You get random colors. And thy dont persist month to month. But they are at least unique, per catwgory
I disagree. The colours are not unique. Starting with nothing, if you add CatA, CatB, and CatC in January you get three unique colours (though some are tonally quite close). If you then move to February and add CatD, it will re-use a colour. The workaround I came up with was to go back to a month well in the past and create all my categories there on dummy appointments. Any time I need to add a new category I have to remember to go back to that month to do it. If I don't, then it will re-use a colour that is not present on the month I am editing.

But... seeing it took me several goes to figure out what on earth it was doing, and the fact you *cannot delete* categories, I now have multiple sets of useless categories that colleagues keep asking me about as they pop up in the selection list.

In short, it is a total mess. I've tried plenty of calendar software — local and online — and never come across a feature quite as broken as this. Actually, broken is probably not the correct word. It's bad design.

**Comment by pabrams** [comment]
Im not saying its good.  I havent run into the data issue you dedcribe, though.  Is thete an issue open somewhere on hpw to reproduce that bug?  If not, it would help.

**Comment by pabrams** [comment]
That paragraph you just wrote starting with "i disagree"  deems to me the opposite, sincce afaict,  youre describing the interaction of the two issues i mentioned ( cannnot choose colors, and colors do not  persist month-to-month)

**Comment by zkarj735** [comment]
> They are unique per montg per category, for me
Find two months in the past that have no entries. On the first month, create an entry with a new category "A". Move to the second month and create an entry with a new category "B". Now go back to the first month and add an entry with category "B". You now have two categories in use on a single month with the same colour.

It sounds like you live in a month, then move to the next one. I was using it to plan out 6 months of work and we spent a considerable amount of time going back and forth between months and thinking of related events that needed to be shown, thus realising we needed more categories.

**Comment by zkarj735** [comment]
And the dual issues are related. Because of the very weird behaviour of categories it is highly likely that spurious categories will result. And those cannot be deleted.

**Comment by pabrams** [comment]
> Find two months in the past that have no entries. On the first month, create an entry with a new category "A". Move to the second month and create an entry with a new category "B". Now go back to the first month and add an entry with category "B". You now have two categories in use on a single month with the same colour.

I haven't tried this yet, but what is the actual result?  You're saying that category A and category B have the same colour in the first month?  Is there an issue created for this bug?

**Comment by pabrams** [comment]
> It sounds like you live in a month, then move to the next one. I was using it to plan out 6 months of work and we spent a considerable amount of time going back and forth between months and thinking of related events that needed to be shown, thus realising we needed more categories.

Yes, perhaps it was an assumption made by the developer, that it wasn't a requirement to hop around adding new categories to past months.  A huge percentage of users would not have that use case.

**Comment by pabrams** [comment]
> nd the dual issues are related. Because of the very weird behaviour of categories it is highly likely that spurious categories will result. And those cannot be deleted.

Hasn't been a problem for me. When it is, I'll write a script to delete the extra months from the data.

**Comment by zkarj735** [comment]
I don't think using a calendar to plan future events is really an edge case. We had to plan our required activities against various restrictions that would dictate when they could not be done, along with some "what if" scenarios that then required us to move stuff around. After getting part way through this process we realised other types of events we had to cater for, hence new categories. 

Understanding the limitations of the calendar ahead of time would have allowed me to avoid the problem of duplicate colours through the silly workaround of reserving a past month for category creation, but in my 30 year career this is not the sort of problem I've ever hit before.

**Comment by melance** [comment]
Having the colors of the categories being set automatically without any way to change them is definitely causing issues.  I need a single category to maintain the same color over time to help reporting be more manageable.  Is this request still being looked at or is it effective dead in the water?

**Comment by pabrams** [comment]
> I don't think using a calendar to plan future events is really an edge case. We had to plan our required activities against various restrictions that would dictate when they could not be done, along with some "what if" scenarios that then required us to move stuff around. After getting part way through this process we realised other types of events we had to cater for, hence new categories.
> 
> Understanding the limitations of the calendar ahead of time would have allowed me to avoid the problem of duplicate colours through the silly workaround of reserving a past month for category creation, but in my 30 year career this is not the sort of problem I've ever hit before.

> I have since discovered that the root cause of this problem is the whole category system in the calendar is seriously broken. It needs tearing down and rebuilding from scratch. I honestly don't know how it ever shipped.

You do know this is free software, right? I'd love to fix it for you, but I'd be working for free, and even though I've asked several times, you haven't even properly described the bug, yet (or if you did, you didn't mention that here).
The fact that you're not willing to do the small amount of work to reproduce the bug and write down the steps doesn't make me want to fix it.
Maybe someone from Microsoft can fix it, or maybe someone who's more generous with their time, than me.

**Comment by pabrams** [comment]
> Having the colors of the categories being set automatically without any way to change them is definitely causing issues. I need a single category to maintain the same color over time to help reporting be more manageable. Is this request still being looked at or is it effective dead in the water?

The last commit was 2 years ago... you tell me

**Comment by JGPease** [comment]
We would also like to see this implemented.

**Comment by Sammi-NZ** [comment]
Any update on this - I would like to change a colour assigned to an event category - I have 3 event categories this month - one, Holidays, is blue the other two, Work Stream A & Work Stream B, are both orange. I'd love to change one Work Stream to a not orange colour.

**Comment by MOlausson** [comment]
We are back to actively maintaining the extension, we're going to triage this one against other work so no promises on when it will be done yet.

**Comment by WillemijnMartens-Visser** [comment]
> Any update on this - I would like to change a colour assigned to an event category - I have 3 event categories this month - one, Holidays, is blue the other two, Work Stream A & Work Stream B, are both orange. I'd love to change one Work Stream to a not orange colour.

Any news on this? I also would like to change colors of the categories that I am using

**Comment by ChemKroonri** [comment]
I like this add-in but please add a way for us to select colors for events.

**Comment by digvijaykroll** [comment]
Any update on this please?? It's been 8 years since this was opened 🤔

**Comment by SvenVandenbrande** [comment]
Any update on this?
Is this project still alive?

**Comment by mattiasen1** [comment]
No ETA currently. The UI revamp should be live in a few weeks at latest.
